#!/usr/bin/env python
PACKAGE = "multi_car"
 
from dynamic_reconfigure.parameter_generator_catkin import *
 
gen = ParameterGenerator()
 
gen.add("vel_kp", double_t, 0, "vel kp parameter", 0.0, 0.0, 10)
gen.add("vel_ki", double_t, 0, "vel ki parameter", 1.5, 0.0, 10)
gen.add("vel_kd", double_t, 0, "vel kd parameter", 0.01, 0.0, 10)

gen.add("theta_kp", double_t, 0, "theta kp parameter", 5.0, 0.0, 10)
gen.add("theta_ki", double_t, 0, "theta ki parameter", 0.1, 0.0, 10)
gen.add("theta_kd", double_t, 0, "theta kd parameter", 0.5, 0.0, 10)

gen.add("pos_kp", double_t, 0, "pos kp parameter", 0.5, 0.0, 10)
gen.add("pos_ki", double_t, 0, "pos ki parameter", 0.0, 0.0, 10)
gen.add("pos_kd", double_t, 0, "pos kd parameter", 0.0, 0.0, 10)
gen.add("k1", double_t, 0, "pos k1", 0.0, 0.0, 10)
gen.add("k2", double_t, 0, "pos k2", 0.0, 0.0, 10)
# Target For Debug
gen.add("vel_x", double_t, 0, "vel_x", 0.0, -1, 1)
gen.add("vel_y", double_t, 0, "vel_y", 0.0, -1, 1)
gen.add("pos_x", double_t, 0, "pos_x", 0.0, -10, 10)
gen.add("pos_y", double_t, 0, "pos_y", 0.0, -10, 10)
gen.add("theta", double_t, 0, "theta", 0.0, 0.0, 1.57)

exit(gen.generate(PACKAGE, "multi_traj_node", "car_param_"))